### DISCUSSION TITLE
New Feature - Enhanced Audit Log to include Descriptions instead of Ids

### DISCUSSION THREAD

**Discussion by ibrahimGit123** [serenity-is/Serenity#7110] [Serenity Ideas]
The Audit Log is a good idea  - https://demo.serenity.is/AdvancedSamples/DataAuditLog

However, it should auto convert Id values into their associated lookup value. In the link above, it shows the Product Id, which is not meaningful. It should include the actual product description. 

![image](https://github.com/serenity-is/Serenity/assets/57414714/4fb019f3-33e5-4bed-ae6a-9e0bc24d0868)

**Comment by VictorTomaili** [comment]
AuditLog is an advanced sample. You need to ask it in StartSharp repository. And no, the value itself is meaningful. If you put the lookup value, each time the value text is changed, the logs will be changed. This is not consistent for the log. It will be store the lookup text as a separated value. But this is also not necessary when you store the lookup table updates in the AuditLog.

**Reply by ibrahimGit123** [reply]
I understand. 

I just think the current audit log is hard to interpret. You can look at other system audit logs as examples for how this is typically done, like Microsoft Dynamics 
https://learn.microsoft.com/en-us/power-platform/admin/manage-dataverse-auditing

Thanks
